test/parse.js

```diff
@@ -65,3 +65,11 @@ test('object properties', t => {
 	t.falsy(fn.parse().prototype);
 	t.deepEqual(fn.parse('hasOwnProperty=foo'), {hasOwnProperty: 'foo'});
 });
+
+test('query strings having indexed arrays', t => {
+	t.deepEqual(fn.parse('foo[0]=bar&foo[1]=baz'), {foo: ['bar', 'baz']});
```

Comment:
Could you swap to `t.deepEqual(fn.parse('foo[1]=bar&foo[0]=baz'), {foo: ['baz', 'bar']});` so that we sure that indexes are respected?